lib/logstash/inputs/gelf.rb

```diff
@@ -68,36 +76,135 @@ def initialize(params)
   public
   def register
     require 'gelfd'
+    @tcp = nil
+    @udp = nil
   end # def register
 
   public
   def run(output_queue)
     begin
-      # udp server
-      udp_listener(output_queue)
+      if @use_tcp
+        if @port_tcp == 0
```

Comment:
We can do the port/port_tcp/port_udp dance in the register block,  so that here in `run` we only need `@port_tcp` and `@port_udp`

Reply:
Done